src/trexio_hdf5.c

```diff
@@ -406,43 +406,76 @@ trexio_exit_code trexio_hdf5_write_nucleus_num(const trexio_t* file, const uint6
   return TREXIO_SUCCESS;
 }
 
-trexio_exit_code trexio_hdf5_read_nucleus_coord(const trexio_t* file, double* coord) {
+trexio_exit_code trexio_hdf5_read_nucleus_coord(const trexio_t* file, double* coord, const int rank, const uint64_t* dims) {
 
   assert (file != NULL);
   assert (coord != NULL);
+  
+  trexio_hdf5_t* f = (trexio_hdf5_t*) file;
+  
+  herr_t status;
+  int rrank;
+  status = H5LTget_dataset_ndims (f->nucleus_group, NUCLEUS_COORD_NAME, 
+                                  &rrank);
+  assert (status >= 0);
```

Comment:
Here, you should  put 
```
if (status < 0) return TREXIO_FAILURE;
```

If HDF5 sets errno here, you can return `TREXIO_ERRNO`.